modules/round/src/main/JsonView.scala

```diff
@@ -36,9 +36,17 @@ final class JsonView(
       g: Game,
       p: GamePlayer,
       user: Option[Either[LightUser.Ghost, User.WithPerf]],
-      withFlags: WithFlags
+      withFlags: WithFlags,
+      showCountryFlags: Boolean
   ): JsObject =
-    Json
+    var countryCode: Option[String] = Some("");
+    if (showCountryFlags) {
+      (user.map {_.toOption.fold(userJsonView.ghost) { u => 
+            countryCode = u.user.profileOrDefault.country
+      } })
+    }
```

Comment:
`user.ifTrue(showCountryFlags).flatMap(_.toOption).map(_.user.profilierOrDefault.country)` is one option to get the country code as an Option. You can just directly pass that to `.add` then, it won't add it if it's `None`, so no need for any `if (..)`.

Reply:
Wooow. Thanks, that was exactly that kind of functional thing I was looking for :o :)

Reply:
Very recent update removing the Either seems to have broken this :x